internal/client/githubclient/githubclient_test.go

```diff
@@ -0,0 +1,357 @@
+//
+// Copyright 2023 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build integration
+
+package githubclient
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/guacsec/guac/internal/client"
+	"github.com/guacsec/guac/internal/testing/testdata"
+)
+
+// Since these are integration tests, we are just going to call an environment variable.
+// This environment variable should also be available to github actions which this project
+// currently uses.
+func getGithubTokenFromEnv() (string, error) {
+	e := os.Getenv("GITHUB_TOKEN")
+	if e == "" {
+		return "", fmt.Errorf("GITHUB_TOKEN empty or unset")
+	}
+	return e, nil
+}
+
+func testGithubClient() *githubClient {
+	ctx := context.Background()
+	t, err := getGithubTokenFromEnv()
+	if err != nil {
+		panic(err)
+	}
+
+	g, err := NewGithubClient(ctx, t)
+	if err != nil {
+		panic(err)
+	}
+
+	return g
+}
+
+func TestNewGithubClient(t *testing.T) {
+	token, err := getGithubTokenFromEnv()
+	if err != nil {
+		panic(err)
+	}
+	type args struct {
+		token string
+	}
+	tests := []struct {
+		name       string
+		args       args
+		wantErr    bool
+		errMessage error
+	}{
+		{
+			name: "valid token for github client",
+			args: args{
+				token: token,
+			},
+			wantErr: false,
+		},
+		{
+			name: "invalid token for github client",
+			args: args{
+				token: "fake token",
+			},
+			wantErr:    true,
+			errMessage: errors.New("GET https://api.github.com/users: 401 Bad credentials []"),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx := context.Background()
+			_, err := NewGithubClient(ctx, tt.args.token)
+			if err != nil {
+				if !tt.wantErr {
```

Comment:
this check should probably be outside, it should also error if it expects and error and doesn't get it